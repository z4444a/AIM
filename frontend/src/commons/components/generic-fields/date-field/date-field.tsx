import React, { ReactNode } from 'react';
import {
  DateFieldConfigModel,
  DateFieldHandlers,
  DateFieldValue,
} from '../models/field-config.model';
import Tooltip from '@material-ui/core/Tooltip';
import moment from 'moment';
import { MaterialUiPickersDate } from 'material-ui-pickers/typings/date';
import AimDatePicker from '../../date-picker/aim-date-picker';
import AimMultipleDatePicker from '../../aim-multiple-date-picker/index';

export interface Props {
  fieldConfig: DateFieldConfigModel;
  handlers: DateFieldHandlers;
  value?: DateFieldValue;
  values?: DateFieldValue[];
  today?: boolean;
  needTooltip?: boolean;
}

export type InternalProps = Props;

export class DateField extends React.PureComponent<InternalProps> {
  public render(): ReactNode {
    const { fieldConfig, value, today, needTooltip, values } = this.props;
    const {
      className,
      style,
      required,
      labelConfig,
      error,
      FormHelperTextProps,
      disabled,
      multipleMax,
    } = fieldConfig;
    let minDate, maxDate;
    if (fieldConfig.constraints) {
      const { minDateValue, maxDateValue } = fieldConfig.constraints;
      minDate = minDateValue;
      maxDate = maxDateValue;
    }
    const multiple = multipleMax && multipleMax > 1;
    const pickerProps = {
      className: className,
      style: style,
      label: labelConfig && labelConfig.visible ? labelConfig.value : undefined,
      required: required,
      error: !!error,
      helperText: error ? error.value : undefined,
      emptyLabel: today ? '"сегодня"' : '',
      minDate: minDate,
      maxDate: maxDate,
      FormHelperTextProps: FormHelperTextProps,
      disabled: today ? true : disabled,
    };
    return (
      <Tooltip
        title={
          needTooltip
            ? fieldConfig.constraints
              ? this.getTooltip(
                  fieldConfig.constraints.minDateValue,
                  fieldConfig.constraints.maxDateValue
                )
              : this.getTooltip()
            : ''
        }
      >
        {multiple ? (
          <AimMultipleDatePicker
            values={values ? values : []}
            valueLimit={multipleMax}
            onValuesChange={this.handleValuesChange}
            value={undefined}
            onChange={() => {}}
            {...pickerProps}
          />
        ) : (
          <AimDatePicker value={value || null} onChange={this.handleValueChange} {...pickerProps} />
        )}
      </Tooltip>
    );
  }

  private handleValuesChange = (values: Date[]) => {
    const { handlers, fieldConfig } = this.props;
    if (!handlers || typeof handlers.onValuesChange !== 'function') {
      return;
    }
    handlers.onValuesChange(values, fieldConfig.key);
  };

  private handleValueChange = (date: MaterialUiPickersDate) => {
    const { handlers, fieldConfig } = this.props;
    if (!handlers || typeof handlers.onValueChange !== 'function') {
      return;
    }
    handlers.onValueChange(date.toDate(), fieldConfig.key);
  };

  private getTooltip = (
    min: Date | undefined = undefined,
    max: Date | undefined = undefined
  ): string => {
    return (
      'Выберите дату' +
      (!min && !max
        ? ''
        : min && max
        ? ' из диапазона ' +
          moment(min).format('YYYY-MM-DD') +
          ' - ' +
          moment(max).format('YYYY-MM-DD')
        : min
        ? '. Дата должна быть не раньше ' + moment(min).format('YYYY-MM-DD')
        : '. Дата должна быть не позже ' + moment(max).format('YYYY-MM-DD'))
    );
  };
}
