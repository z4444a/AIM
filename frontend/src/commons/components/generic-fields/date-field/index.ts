import { DateField } from './date-field';

export default DateField;
