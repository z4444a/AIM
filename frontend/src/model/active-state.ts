export const ActiveState = [
  {
    key: '1',
    value: 'Активен',
  },
  {
    key: '2',
    value: 'Не активен',
  },
];
