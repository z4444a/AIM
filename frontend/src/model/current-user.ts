import { Role } from '../commons/role';

export interface CurrentUser {
  id: number;
  fullName: string;
  username: string;
  role: Role;
}
