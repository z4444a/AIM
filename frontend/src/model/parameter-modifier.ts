export enum ParameterModifier {
  POOL_PARAMETER = 1,
  REQUEST_PARAMETER = 2,
  ALLOCATION_PARAMETER = 3,
}
