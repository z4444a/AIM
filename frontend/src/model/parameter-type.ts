export enum ParameterType {
  NUMBER = 'NUMBER',
  TEXT = 'TEXT',
  DATE = 'DATE',
  LIST = 'LIST',
  REAL = 'REAL',
  POOL = 'POOL',
}
