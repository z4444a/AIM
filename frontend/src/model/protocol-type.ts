export const Protocols = [
  {
    key: '1',
    value: 'TLS',
  },
  {
    key: '2',
    value: 'SSL',
  },
  {
    key: '3',
    value: 'None',
  },
];
