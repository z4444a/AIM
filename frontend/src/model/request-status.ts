import { RequestStatus } from './get/request-status';

export function RequestStatuses(t: (s: string) => string) {
  return [
    {
      key: RequestStatus.NEW.toString(),
      value: t('requestPage.status.new'),
    },
    {
      key: RequestStatus.IN_PROGRESS.toString(),
      value: t('requestPage.status.inWork'),
    },
    {
      key: RequestStatus.PROCESSED.toString(),
      value: t('requestPage.status.executed'),
    },
    {
      key: RequestStatus.CANCELED.toString(),
      value: t('requestPage.status.canceled'),
    },
    {
      key: RequestStatus.PAUSED.toString(),
      value: t('requestPage.status.paused'),
    },
  ];
}
