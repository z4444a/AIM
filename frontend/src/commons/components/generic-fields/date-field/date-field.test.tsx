import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { DateField, InternalProps } from './date-field';
import { DateFieldConfigModel, DateFieldHandlers } from '../models/field-config.model';

describe('DateField', () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    const fieldConfig = new DateFieldConfigModel('dateField');
    const handlers: DateFieldHandlers = {
      onValueChange: () => {},
    };

    const props: InternalProps = {
      fieldConfig,
      handlers,
    };
    wrapper = shallow(<DateField {...props} />);
  });

  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
