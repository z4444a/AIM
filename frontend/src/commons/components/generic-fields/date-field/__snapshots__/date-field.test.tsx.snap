// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateField should render correctly 1`] = `
<WithStyles(Tooltip)
  title=""
>
  <AimDatePicker
    emptyLabel=""
    error={false}
    onChange={[Function]}
    required={false}
    value={null}
  />
</WithStyles(Tooltip)>
`;
